### PR TITLE
Updates to the configuration check utility for Python 2.7

### PR DESCRIPTION
**Pull request by connersaeli**
This PR is the Python 2.7 equivalent to [this PR](https://github.com/ubccr/supremm/pull/257).

### CODE REVIEW

File: src/supremm/supremmconf.py [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
+    opts, _ = getopt(sys.argv[1:], "dc:s:i:h", ["debug", "config=", "section=", "item="])
 
     for opt in opts:
+        if opt[0] in ("-d", "--debug"):
+            retdata['log'] = logging.DEBUG 
+        if opt[0] in ("-c", "--config"):
+            retdata['config'] = opt[1]
         if opt[0] in ("-s", "--section"):
-            retdata['section'] = opt[1].encode("utf-8")
+            retdata['section'] = opt[1]
```
Comment (jpwhite4) [inline src/supremm/supremmconf.py:40]: How come the `encode("utf-8")` function has been removed?

**Review by jpwhite4 — CHANGES_REQUESTED** [review]

File: src/supremm/supremmconf.py (hunk 1 above)
Comment (connersaeli) [reply src/supremm/supremmconf.py:40]: When I was working with the python3 version this script wasn't able to read my configuration unless that was removed. I carried over all the changes from the py3 updates to this PR, so it might not be necessary for this version of the software.

Comment (connersaeli) [reply src/supremm/supremmconf.py]: Just added that back in

File: src/supremm/supremmconf.py [hunk 2]
```diff
@@ -41,18 +53,44 @@ def getoptions():
...
+        try:
+            item = section[opts['item']]
+        except KeyError:
+            logging.error("Item '{}' not defined in section '{}'.".format(opts['item'], opts['section']))
+            sys.exit(1)
+
+        if isinstance(item, dict):
+            item = json.dumps(item, indent=4)
+ 
+        print("{} : {}".format(opts['item'], item))
```
Comment (jpwhite4) [inline src/supremm/supremmconf.py:90]: This looks like a breaking change. The old code would print out the requested item, whereas the new code prints out the item name and the requested item separated by " : ".

What is the reason for the change and what is the impact of this change?

File: src/supremm/supremmconf.py [hunk 3]
```diff
@@ -1,29 +1,41 @@
...
     """ print usage """
-    print "usage: {0} [OPTS]".format(os.path.basename(__file__))
-    print "  -s --section SECTION  output the configuration data from the specified section"
-    print "  -i --item ITEM        output the configuration data for the specified item"
-    print "  -h --help             print this help message"
-
+    print("usage: {0} [OPTS]".format(os.path.basename(__file__)))
+    print("  -d --debug            set log level to debug")
+    print("  -c --config           specify the configuration file")
+    print("  -r --resources        output the configuration data for all enabled resources")
```
Comment (jpwhite4) [inline src/supremm/supremmconf.py:19]: There doesn't appear to be any code that implements the `-r` flag that is listed here. 

Also the help text for the `-c` option is misleading - you need to specify the path to the directory that contains the conifguration file and not the file path.

**Review by jpwhite4 — CHANGES_REQUESTED** [review]

File: src/supremm/supremmconf.py [hunk 4]
```diff
@@ -1,29 +1,41 @@
...
-    print "  -i --item ITEM        output the configuration data for the specified item"
-    print "  -h --help             print this help message"
-
+    print("usage: {0} [OPTS]".format(os.path.basename(__file__)))
+    print("  -d --debug            set log level to debug")
+    print("  -c --config           specify the configuration file")
+    print("  -r --resources        output the configuration data for all enabled resources") 
+    print("  -s --section SECTION  output the configuration data from the specified section")
+    print("  -i --item ITEM        output the configuration data for the specified item")
+    print("  -h --help             print this help message")
```
Comment (jpwhite4) [inline src/supremm/supremmconf.py:22]: Also `--help` is not a valid option. The current code only lets you use `-h`. Suggest that you add the `--help` as a valid alias of `-h` (even though it was not in the scope of the original set of changes.)

File: src/supremm/supremmconf.py (hunk 2 above)
Comment (connersaeli) [reply src/supremm/supremmconf.py:90]: Sorry, I didn't realize this is for the python2 PR. That is the python3 version

File: src/supremm/supremmconf.py (hunk 1 above)
Comment (connersaeli) [reply src/supremm/supremmconf.py]: Just thought I'd mention that the utf-8 encoding was removed in the py3 version because of this error. Top screenshot is with encoding and bottom screenshot is without encoding.
image
image
The code works with or without the utf-8 encoding for py2 so my thought process was to just remove it altogether.

File: src/supremm/supremmconf.py (hunk 4 above)
Comment (connersaeli) [reply src/supremm/supremmconf.py:22]: Added alias for --help.

File: src/supremm/supremmconf.py (hunk 3 above)
Comment (connersaeli) [reply src/supremm/supremmconf.py:19]: Updated the misleading text. As for the -r flag, I think I had an idea to list individual resources but forgot about it. Seeing that it's the same as `python supremmconf.py -s resources` I've just removed it.

File: src/supremm/supremmconf.py (hunk 2 above)
Comment (connersaeli) [reply src/supremm/supremmconf.py:90]: Just tried and it works the same for py2

**Review by jpwhite4 — APPROVED** [review]

**Comment by jpwhite4** [comment]
After you merge this please can you cherry-pick (or equivalent) this up to the 2.0 branch
